salt/utils/event.py

```diff
@@ -200,7 +200,16 @@ def tagify(suffix='', prefix='', base=SALT):
         parts.extend(suffix)
     else:  # string so append
         parts.append(suffix)
-    return TAGPARTER.join([part for part in parts if part])
+
+    for index, _ in enumerate(parts):
+        if isinstance(parts[index], six.integer_types):
+            parts[index] = str(parts[index])
+        else:
+            try:
+                parts[index] = salt.utils.stringutils.to_str(parts[index])
+            except TypeError:
+                pass
+    return TAGPARTER.join(parts)
```

Comment:
This does a type check on each part. I don't see any reason to do that, to be honest. What if the list item isn't an int or string? That'd _really_ cause a problem when we go to join it, no? How about this?

```python
    for index, _ in enumerate(parts):
        try:
            parts[index] = salt.utils.stringutils.to_str(parts[index])
        except TypeError:
            parts[index] = str(parts[index])
    return TAGPARTER.join(parts)
```

Reply:
Yes, I've updated it to use this version.